lib/logmatcher.c

```diff
@@ -414,6 +417,26 @@ log_matcher_pcre_re_feed_backrefs(LogMatcher *s, LogMessage *msg, gint value_han
     }
 }
 
+static inline void
+log_matcher_pcre_re_feed_value_by_name(LogMatcherPcreRe *s, LogMessage *msg, GString *formatted_name, gchar *tabptr,
+                                       const gchar *value, gint begin_index, gint end_index)
+{
+  if(s->prefix != NULL)
+    {
+      if (formatted_name->len > 0)
+        g_string_truncate(formatted_name, s->prefix_len);
```

Comment:
formatted_name can be a newly allocated GString instance, so truncating may not produce the required prefix. you will need to assign a new value every time.

Reply:
I just recognized that there's a similar pattern in kv-parser(), which seems to be buggy as well. The scratch-buffers allocator does not guarantee that we get the same GString instance every time (though it may happen). Can you fix that part too?

Reply:
> formatted_name can be a newly allocated GString instance, so truncating may not produce the required prefix. you will need to assign a new value every time.

Hi, bazsi, I have a question. In the `scratch_buffers_alloc` function, the allocated buffer will be truncated to 0.
```c++
// lib/scratch-buffers.c, lines 127-128
GString *buffer = g_ptr_array_index(scratch_buffers, scratch_buffers_used);
g_string_truncate(buffer, 0);
```
In this case, if formatted_name is a newly allocated GString instance, it will go to the else case and call `g_string_assign(formatted_name, s->prefix);`.

Reply:
you are right. I missed the loop on the regexp values, with scratch_buffers_alloc() before the loop it all makes sense now.

sorry.